framework/include/base/Assembly.h

```diff
@@ -434,10 +438,29 @@ class Assembly
   const Node * const & nodeNeighbor() const { return _current_neighbor_node; }
 
   /**
-   * Creates the volume, face and arbitrary qrules based on the orders passed in.
+   * Creates volume, face and arbitrary qrules based on the orders passed in
+   * that apply to all subdomains.
    */
   void createQRules(QuadratureType type, Order order, Order volume_order, Order face_order);
 
+  /**
+   * Creates block-specific volume, face and arbitrary qrules based on the
+   * orders passed in.  Any quadrature rules specified using this function
+   * override those created via in the non-block-specific/global createQRules
+   * function.
+   */
+  void createQRules(
+      QuadratureType type, Order order, Order volume_order, Order face_order, SubdomainID block);
+
+  /**
+   * Increases the elemennt/volume quadrature order for the specified mesh
+   * block if and only if the current volume quadrature order is lower.  This
+   * can only cause the quadrature level to increase.  If volume_order is
+   * lower than or equal to the current volume/elem quadrature rule order,
+   * then nothing is done (i.e. this function is idempotent).
+   */
+  void bumpVolumeQRuleOrder(Order volume_order, SubdomainID block);
```

Comment:
Why do we need two functions, this and the above to set orders?

Reply:
Because you don't know what the orders may have been set to.  And the above functions recreate *all* the quadrature rules.  If you just want to bump volume quadrature, you would have to know/re-specify the face and general orders again to be what they were - except you don't know what they were.